### PR TITLE
Windows: Refactor Process and MountLabel

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

Tackling more "TODO Windows" annotations. This factors out ProcessLabel and MountLabel which aren't relevant in the Windows daemon. Have run test-integration-cli and unit-test locally with no regressions.

### CODE REVIEW

**Comment by tiborvass** [comment]
LGTM

**Comment by tiborvass** [comment]
@jhowardmsft please let us know if you need this (or other PRs) to be on one of the windows milestones. TP4 or TP5

**Comment by jhowardmsft** [comment]
Of course :) This is just a general change/improvement though.

File: daemon/container_windows.go [hunk 1]
```diff
@@ -202,3 +200,9 @@ func (daemon *Daemon) mountVolumes(container *Container) error {
 func (container *Container) unmountVolumes(forceSyscall bool) error {
 	return nil
 }
+
+// reserveLabel reserves a label for the process label in the container. This
+// is a no-op on Windows
+func reserveLabel(container *Container) error {
```
Comment (LK4D4) [inline daemon/container_windows.go]: Hmm `label.ReserveLabel` already should be no-op for `!linux`:
https://github.com/opencontainers/runc/blob/master/libcontainer/label/label.go#L43

File: daemon/delete_unix.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// +build linux freebsd
+
+package daemon
+
+// platformSpecificRm is a platform-specific helper function for rm/delete
+func platformSpecificRm(container *Container) {
```
Comment (LK4D4) [inline daemon/delete_unix.go]: It really doesn't sound like it(`platformSpecificRm`).

File: daemon/inspect_unix.go [hunk 3]
```diff
@@ -75,3 +78,25 @@ func addMountPoints(container *Container) []types.MountPoint {
 	}
 	return mountPoints
 }
+
+// populateJSONBase is a platform-specific helper function for inspect that
+// populates the ContainerJSONBase structure
+func populateJSONBase(container *Container, hostConfig *runconfig.HostConfig, containerState *types.ContainerState) *types.ContainerJSONBase {
+
```
Comment (LK4D4) [inline daemon/inspect_unix.go]: What this empty lines? :) However I propose to create smaller function like `setLabels`, it'll be more error-prone to changing structs later(and it's changing often).

**Comment by LK4D4** [comment]
ping @jhowardmsft :)

**Comment by jhowardmsft** [comment]
@LK4D4 Dealing with last minute TP4 fires - haven't forgotten :)

**Comment by tiborvass** [comment]
@jhowardmsft needs rebase sorry :(

**Comment by jhowardmsft** [comment]
Yeah, might be Jan now - on vacation for a while... :)

**Comment by thaJeztah** [comment]
@tiborvass @LK4D4 should somebody carry this PR then?

**Comment by tiborvass** [comment]
@thaJeztah yes we'll carry it

**Comment by jhowardmsft** [comment]
No need to carry. I'll take care of it when I can. It's not critical, just a tidy up. Just haven't got time right now to concentrate on it while on vacation - working on getting a robust Windows to Linux CI has taken precedence beyond anything else, plus adding Windows to Windows CI.

**Comment by tiborvass** [comment]
@jhowardmsft if you don't want us to carry it, would you mind closing the PR? You can reopen before pushing a rebased version.

**Comment by thaJeztah** [comment]
ping @jhowardmsft, now that you're back, could you rebase this so that we can merge?

**Comment by thaJeztah** [comment]
ping @jhowardmsft what's the status here? Can you address the comments and rebase?

**Comment by jhowardmsft** [comment]
Closing while busy on CI. Will revisit.

**Pull request closed without merging**